examples/ping-pong-encrypted/src/client.rs

```diff
@@ -0,0 +1,77 @@
+use crate::messages::{Message, Ping, Pong, PING_MSG_TYPE, PONG_MSG_TYPE};
+use codec_sv2::{Frame, HandshakeRole, Initiator, StandardSv2Frame};
+use key_utils::Secp256k1PublicKey;
+use network_helpers_sv2::noise_connection_tokio::Connection;
+use tokio::net::TcpStream;
+
+use crate::error::Error;
+
+pub async fn start_client(address: &str, k_pub: String) -> Result<(), Error> {
+    let stream = TcpStream::connect(address).await?;
+
+    println!("CLIENT: Connected to server on {}", address);
+
+    // parse server pubkey
+    let k_pub: Secp256k1PublicKey = k_pub.try_into()?;
+
+    // noise handshake initiator
+    let initiator = Initiator::from_raw_k(k_pub.into_bytes())?;
+
+    // channels for encrypted connection
+    let (receiver, sender, _, _) =
+        Connection::new(stream, HandshakeRole::Initiator(initiator)).await?;
+
+    // create Ping message
+    let ping = Ping::new()?;
+    let ping_nonce = ping.get_nonce();
+    let message = Message::Ping(ping);
+
+    // create Ping frame
+    let ping_frame =
+        StandardSv2Frame::<Message>::from_message(message.clone(), PING_MSG_TYPE, 0, false)
```

Comment:
https://stratumprotocol.org/specification/09-Extensions/
Is a 0 extension type unique in some way?

Reply:
since there are no extensions defined yet, using `0` is just the void/null case